### PR TITLE
[Spot] Fix memory contraint for spot controller and backward compatibility for spot commands

### PR DESCRIPTION
**Pull request by Michaelvll**
Fix spot controller memory contraint.
This is a bug introduced by #3191.

This also fixes the issue with the job_lib assertion for spot jobs. To reproduce:
1. `git checkout 053d0ba2b58705c3bc5540729eb5084911e7fa9a`; `sky spot launch -n test-long-run --cloud gcp --cpus 2 -d "echo hi; sle
ep 100000000000000"`
2. `sky checkout master`
3. `sky spot launch -n test-1 echo hi`
4. `sky spot logs 1` fails due to the assertion

After this PR, it should fix this backward compat issue for the following two cases:
1. if the original commit is before 053d, it should work normally with the current PR
2. if the original commit is the current master, and the spot controller is updated to the current master as the repro script above, after `sky start sky-spot-controller-<hash>`, all the commands should work normally.

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
  - [x] `sky spot launch -n test echo hi` without a controller
  - [x] Reproducible code above.
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/resources.py [hunk 1]
```diff
@@ -448,7 +448,9 @@ def _set_memory(
 
         self._memory = str(memory)
         if isinstance(memory, str):
-            if memory.endswith('+'):
+            if memory.endswith(('+', 'x')):
+                # 'x' is used internally for make sure our resources used by
```
Comment (concretevitamin) [inline sky/resources.py:452]: add example e.g., 3x

File: sky/spot/constants.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # OOM (each vCPU can have 4 spot controller processes as we set the CPU
 # requirement to 0.25, and 3 GB is barely enough for 4 spot processes).
 # We use 50 GB disk size to reduce the cost.
-CONTROLLER_RESOURCES = {'memory': '3x', 'disk_size': 50}
+CONTROLLER_RESOURCES = {'cpus': '8+', 'memory': '3x', 'disk_size': 50}
```
Comment (concretevitamin) [inline sky/spot/constants.py:16]: Update L9-11?

File: sky/resources.py
Comment (concretevitamin) [inline sky/resources.py:1]: Can we add a unit test to parse this?

Q: where does the "multipler" logic get executed?

**Review by concretevitamin — APPROVED** [review]

File: sky/spot/constants.py (hunk 2 above)
Comment (concretevitamin) [reply sky/spot/constants.py:16]: Oh, I meant that those lines suggest concrete instance types, and here we do not necessarily use those? E.g., those VMs have 32GB, and the spec on this line suggests >= 24GB?

Comment (Michaelvll) [reply sky/spot/constants.py:16]: Updated the comments a little bit :)

**Review by concretevitamin — APPROVED** [review]
LGTM

**Pull request merged**